src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -908,6 +908,12 @@ public async Task UpdateLicenseAsync(Guid organizationId, OrganizationLicense li
 
         public async Task DeleteAsync(Organization organization)
         {
+            var ssoConfig = await _ssoConfigRepository.GetByOrganizationIdAsync(organization.Id);
+            if (ssoConfig?.GetData()?.UseKeyConnector == true)
+            {
+                throw new BadRequestException("You cannot delete an Organization that is using Key Connector.");
+            }
+
```

Comment:
Lets move this logic to a private method & preferabbly write a quick unit test for it. Also: why can't you delete an org that is using key connector?

Reply:
It's a quick way to prevent locking out all the users using key connector.

Reply:
Added tests.